### ISSUE TITLE
This is only useful for top-level packages

### ISSUE THREAD

**Issue by moos** [rogeriochaves/npm-force-resolutions#12]
Per [package-lock.json](https://docs.npmjs.com/configuring-npm/package-lock-json.html):
> One key detail about package-lock.json is that it cannot be published, and it will be ignored if found in any place other than the toplevel package.

Meaning this will NOT work for nested packages.  [yarn's](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/) has a notice "Nested packages may not work properly" under Caveats.

It'd be good to add such a (large!) caveat prominently to the README.

**Comment by Flashfyre** [comment]
Yeah, this would be a perfect solution to my nested dependencies issue if it weren't for the fact that it doesn't remove the version from the 'requires' section of my top level dependencies.

**Comment by Flashfyre** [comment]
I ended up switching to Yarn which has native resolution support. I also installed Yalc to handle linking to my local versions for testing (NPM/Yarn links didn't work due to the way they were handled). This solved my problem.

**Comment by danmarshall** [comment]
For me, I now invoke it manually instead of `preinstall`.

**Comment by vit100** [comment]
What if you do manually or via script `npx npm-force-resolutions`, it will update your `package-lock.json` and then you do `npm shrinkwrap`. And shrinkwrap file (basically it is same as package-lock.json) you can push into repo.
